qiskit/pulse/commands/frame_change.py

```diff
@@ -34,42 +28,23 @@ def __init__(self, phase: float, name: Optional[str] = None):
             phase: Frame change phase in radians. The allowable precision is device specific
             name: Name of this framechange command.
         """
-        super().__init__(duration=0)
-        self._phase = float(phase)
-        self._name = FrameChange.create_name(name)
-
-    @property
-    def phase(self):
-        """Framechange phase."""
-        return self._phase
-
-    def __eq__(self, other: 'FrameChange'):
-        """Two FrameChanges are the same if they are of the same type and phase.
-
-        Args:
-            other: other FrameChange
-
-        Returns:
-            bool: are self and other equal
-        """
-        return super().__eq__(other) and (self.phase == other.phase)
-
-    def __hash__(self):
-        return hash((super().__hash__(), self.phase))
+        warnings.warn("FrameChange is deprecated. Use ShiftPhase, instead, with channels specified. "
+                      "For example: ShiftPhase(3.14)(DriveChannel(0)) -> "
+                      "ShiftPhase(3.14, DriveChannel(0)). Returning a ShiftPhase instance. This can "
+                      "be called to add a channel: ShiftPhase(3.14)(DriveChannel(0)).",
+                      DeprecationWarning)
+        self = ShiftPhase(phase, name=name)
```

Comment:
`-phase`? I assume u1(theta) = FrameChange(-theta) = ShiftPhase(theta), so this refactor would be more than rename. (I prefer `ShiftPhase` with same sign with u1.) I couldn't find any mention to sign inversion in the RFC, so maybe this is just my misunderstanding. Are you intending just rename here? @taalexander

Reply:
@itoko why do you think it should be `-phase`? I must be missing something, I don't know where the negative sign is coming from

Reply:
@lcapelluto thank you for your hard work on the series of cmd/inst refactoring. I found the negative sign in `cmd_def` (and so in `instruction_schedule_map`).
```
backend = provider.get_backend('ibmq_armonk')  
backend.defaults().cmd_def
->
Command(name='u1', qubits=[0], sequence=[PulseQobjInstruction(ch='d0', name='fc', phase='-(P0)', t0=0)]),

backend.defaults().instruction_schedule_map.get('u1', 0, 1.57)              
->
Schedule((0, FrameChangeInstruction(FrameChange(phase=-1.570, name="fc0"), DriveChannel(0))), name="u1")
```
This may imply u1(theta) = FrameChange(-theta). So currently user must do FrameChange(-pi/2) if they want to do u1(pi/2) or rz(pi/2). I thought we're going to remove this negative sign from the frontend user in the new `ShiftPhase` because `ShiftPhase` sounds like changing the phase of the qubit itself as well as u1/rz while `FrameChange` changes the frame of the qubit as its name suggests. (I know changing qobj spec is difficult, so we may still need to use "fc" in qobj, that means we need to flip sign when assembling.)

Reply:
I think negative sign comes from difference of positive theta definition (clockwise or counter clockwise rotation direction) in somewhere between logical/physical layer.

Reply:
Seems to make sense at the moment that a positive z rotation by angle theta on a qubit would require a phase shift by negative theta. Shift phase is shifting the phrase tracking the frame. The frame is both phase and frequency, that is why we are moving to set/shift phase and set/shift frequency

Reply:
OK, I see. Since the `ShiftPhase` takes `Channel` (not `Qubit`), it should mean *shift output signal phase of the channel* (not shift phase of the qubit as I misunderstood). I'm not sure how many people make a mistake like me, but mentioning to the relation `rz(theta) on Qubit(i) = ShiftPhase(-theta, PulseChannel(i))` somewhere in the docstring  would be helpful for novice users.